src/HTSPTypes.h

```diff
@@ -91,17 +92,27 @@ enum eHTSPEventType
 struct SHTSPEvent
 {
   eHTSPEventType m_type;
-  uint32_t       m_idx;
 
-  SHTSPEvent (eHTSPEventType type = HTSP_EVENT_NONE, uint32_t idx = 0) :
+  // params for HTSP_EVENT_EPG_UPDATE
+  tvheadend::entity::Event m_epg;
+  EPG_EVENT_STATE          m_state;
+
+  SHTSPEvent (eHTSPEventType type = HTSP_EVENT_NONE) :
     m_type(type),
-    m_idx (idx)
+    m_state(EPG_EVENT_CREATED)
   {
   }
-  
+
+  SHTSPEvent (eHTSPEventType type, const tvheadend::entity::Event &epg, EPG_EVENT_STATE state) :
```

Comment:
With the code from #161 this special case event could be subclassed from the more generic base event which only needs a type member.